scipy/interpolate/_bsplines.py

```diff
@@ -314,6 +322,14 @@ def __call__(self, x, nu=0, extrapolate=None):
         x = np.asarray(x)
         x_shape, x_ndim = x.shape, x.ndim
         x = np.ascontiguousarray(x.ravel(), dtype=np.float_)
+
+        # With periodic extrapolation we map x to the segment
+        # [self.t[k], self.t[-k]].
+        if extrapolate == 'periodic':
+            x = self.t[self.k] + (x - self.t[self.k]) % (self.t[-(self.k)] -
```

Comment:
I think it should be `self.t.size - self.k - 1` since this would return `n` (`t[n]` being the upper limit of the base interval). Also, if `k=0` then `self.t.size - self.k` would return `n+1` which is an out-of-bounds index for `t`.